next/blog/pages/index.tsx

```diff
@@ -0,0 +1,38 @@
+import { useEffect } from 'react';
+import {GetServerSideProps} from 'next';
+import { APIResponse, HomePageData } from '~~/types';
+import { useHeadTags } from '~/composables/og-head';
+import {PageWrapper} from "~/components/PageWrapper";
+import { HomeApi} from "~/api";
+import {HomepageHero} from "~/components/home-page/Hero";
+import {HomePageBlogsList} from "~/components/home-page/BlogList";
+
+export default function HomePage (props: APIResponse<HomePageData>): JSX.Element {
+  const { setOgHead } = useHeadTags();
+  useEffect(() => {
+    setOgHead({
+      title: props.data?.meta?.title,
+    });
+  }, [props.data]);
+
+  return (
+      <PageWrapper header={props.header} footer={props.footer}>
+        {props.data !== null  && (
+           <>
+             <HomepageHero data={props.data.meta.hero} />
+             <HomePageBlogsList blogs={props.data.blogs} />
+           </>
+        )}
+      </PageWrapper>
+  );
+}
+
+export const getServerSideProps: GetServerSideProps<APIResponse<HomePageData>> = async () => {
```

Comment:
You should use SSG, `getServerSideProps` is used in SSR. For more info. see: https://nextjs.org/docs/pages/building-your-application/rendering/static-site-generation

Reply:
@siradji